subprojects/launcher/src/integTest/groovy/org/gradle/launcher/continuous/SmokeContinuousIntegrationTest.groovy

```diff
@@ -300,9 +417,11 @@ class SmokeContinuousIntegrationTest extends AbstractContinuousIntegrationTest {
     def "failure to determine inputs cancels build and has a reasonable message after initial success"() {
         when:
         def bFlag = file("bFlag")
+        file("inputA").createFile()
         buildScript """
             task a {
                 inputs.files file("inputA")
+                outputs.files "build/outputA"
```

Comment:
Why do we need to declare outputs now and not before?

Reply:
Because we only watch what we snapshot, and for tasks without outputs we don't snapshot anything.